sos/archive.py

```diff
@@ -532,9 +535,24 @@ def cleanup(self):
         if os.path.isdir(self._archive_root):
             shutil.rmtree(self._archive_root)
 
+    def add_final_manifest_data(self, method):
+        """Adds component-agnostic data to the manifest so that individual
+        SoSComponents do not need to redundantly add these manually
+        """
+        end = datetime.now()
+        start = self.manifest.start_time
+        run_time = end - start
+        self.manifest.add_field('end_time', end)
+        self.manifest.add_field('run_time', run_time)
+        self.manifest.add_field('compression', method)
+        self.add_string(self.manifest.get_json(indent=4),
+                        os.path.join('sos_logs', 'manifest.json'))
+
     def finalize(self, method):
         self.log_info("finalizing archive '%s' using method '%s'"
                       % (self._archive_root, method))
+        if self.manifest is not None:
+            self.add_final_manifest_data(method)
```

Comment:
No metadata are written when `--build` option is used. Since this `finalize` method is called only when `--build` isnt set (cf. https://github.com/sosreport/sos/blob/master/sos/report/__init__.py#L1099-L1106).

A possible fix is to call at https://github.com/sosreport/sos/blob/master/sos/report/__init__.py#L1122 either `self.archive.add_final_manifest_data(None)` directly, or `self.archive.finalize(None)` (that would require some updates in `finalize` to skip compression when `method=None`).

Reply:
Yeah, my hope was that putting it in `archive.finalize()` was to "automagic" it for any components that write one, but I overlooked this.

But, at the same time not a big issue to make the call in each component.